archive/tar_windows.go

```diff
@@ -101,3 +141,310 @@ func setxattr(path, key, value string) error {
 	// since xattrs should not exist in windows diff archives
 	return errors.New("xattrs not supported on Windows")
 }
+
+func prepareApply() error {
```

Comment:
Could these privilege calls be done by the caller?

Reply:
Certainly could be. Moving it to the client would also mean that the `archive.Apply` would not need to be aware if it is running isolated or not, which is likely better.